### DISCUSSION TITLE
CORS errors on a subset of images?

### DISCUSSION THREAD

**Discussion by jensr** [biigle/core#574] [Q&A] [answered]
Hello
I am not really a Biigle user, but trying to support a colleague who is using it and is getting CORS errors after we set up a remote volume with video and images for annotation. I've been through the volumes (2) and tested that they support the recommended CORS settings (GET, OPTIONS etc.), but my colleague is now getting CORS errors only on some images within the volume. 
The CORS settings are defined in a htaccess file on an apache server with header mod enabled.  With me not being terribly familiar with Biigle, I was wondering if anyone has suggestions to how a subset of images can throw an error related to CORS when the settings are defined universally for the whole folder?
Thank you in advance :)

**Comment by mzur** [comment]
Hi, can you please send me the link to an affected volume? BIIGLE can only guess if CORS is configured correctly and maybe there is a different issue with some of the images that makes BIIGLE think it's because of CORS.

**Comment by jensr** [comment]
Hi @mzur - thanks for getting back so quick. 
The volume/location giving up trouble is on https://marine.gov.scot/datafiles/video/20201005200844/
I tested it with cors-test.codehappy.dev and got positive test returns for both GET and OPTIONS calls, but appreciate you taking a look!

**Reply by mzur** [reply]
Thanks. I browsed through the whole volume in BIIGLE and didn't get any CORS error. Does your colleague always get the error for the same image(s) or each time for different ones? Maybe it is a (temporary) network issue?

**Reply by jensr** [reply]
Thank you very much for checking - She was getting a persistent error on a few images, but different experiences on different computers. So perhaps it would be a network issue. 
I very much appreciate your quick response and taking the time to check!

**Reply by RBoschen-Rose** [reply]
Hi Martin, I am the colleague Jens mentions in the message above and am unfortunately still seeing the CORS error for a subset of images.
From one work laptop, the error comes up for images #256, 279 & 280, from another work laptop the error occurs for images #173, 211, 279, 286, 321 & 1904. From my personal laptop (on the same WiFi connection as the two laptops above) I do not have any CORS errors for any images in the folder.
Jens and I have done some tests and we don't believe it is a network issue. The issue is persistent, always for the same images and always from those two machines. Re-booting the laptops, shutting Biigle, logging in and out of Biigle, removing the volume and adding it again unfortunately doesn't solve the problem.
Could you please advise what we should be doing next to troubleshoot the issue? Any help on this would be greatly appreciated! Thanks.

**Reply by mzur** [reply]
Please open the browser developer tools (<kbd>F12</kbd>), select the "Console" tab and look for errors there when you try to open the problematic images. Then please post the errors here.

**Comment by RBoschen-Rose** [comment, marked as answer]
Hi Martin, thanks for your message. We have literally just resolved the issue by deleting the Edge browser history on both of my work laptops. I don’t know why it worked but it did! Worth noting for any other users that have erratic / persistent ‘CORS’ issues with specific images. These may not really be CORS issues, it could be something in the browser history causing problems. Thanks for your help!

**Reply by mzur** [reply]
Please also note that BIIGLE only officially supports Firefox and Chrome.

**Comment by RBoschen-Rose** [comment]
Thanks, that is good to know. I do have the option of using Chrome on one of my work laptops so will conduct my work in Biigle there.
